### DISCUSSION TITLE
Read files from a Golang serverless function

### DISCUSSION THREAD

**Discussion by BlueSlimee** [vercel/vercel#6090] [Help]
Hey there!
It'd be very useful if serverless functions running golang could access files just like [node.js functions can](https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions).

I need to read from a TTF file but my application can't find it. I tried everything but nothing worked. I even tried reading all files in the current directory to see if I was using the wrong path, but the folder that I added to my `vercel.json` file wasn't there, so it's probably a runtime limitation (since it works with node.js functions just fine).

Thanks!

**Comment by smcelhinney** [comment]
I've noticed there's very little in the way of support/articles for this runtime. I'm facing the same issues above, and I used the same approach as above (adding `includeFiles` to my `vercel.json`), but to no avail. If I output `os.Getwd()` in development mode it gives me the path to the build cache dir, so I suspected that it would do something similar when deployed, and I could use `path/filepath.Abs` to get the absolute path to the build directory, but again, no joy. 

Any help on this would be greatly appreciated

**Reply by barelyhuman** [reply]
@smcelhinney @BlueSlimee  
https://github.com/vercel/vercel/discussions/6316#discussioncomment-901041
this might help for simple use cases
